.gitignore

```diff
@@ -3,10 +3,8 @@ dist/
 *.d
 private/
 *.properties
-*.xml
 *.mk
 *.bash
-Makefile
```

Comment:
It is... The relative path is stored in git structure, but Makefiles use absolute path. This is true for make on any system. 

You can try it by copying and pasting it into a new folder, the old makefile won't work and will provide an error.